### PR TITLE
JohnPapa Style Guide : Component files

### PR DESCRIPTION
**Pull request by wmarques**
Continuing JohnPapa implementation as suggested in #2271. 
Refactored components module, spec and e2e tests are OK.

### CODE REVIEW

File: generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    vm.alerts.push(
+                    AlertService.add(
+                {
+                    type: 'danger',
+                    msg: key,
+                    params: data,
+                    timeout: 5000,
+                    toast: AlertService.isToast(),
+                    scoped: true
+                },
```
Comment (mraible) [inline generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js]: This block looks like it should be indented by 8 spaces.

File: generators/client/templates/src/main/webapp/app/components/language/_language.constants.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+(function () {
+    'use strict';
+    
+    angular
+        .module('<%=angularAppName%>')
+    /*
+     Languages codes are ISO_639-1 codes, see http://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
+     They are written in English to avoid character encoding issues (not a perfect solution)
+     */
```
Comment (mraible) [inline generators/client/templates/src/main/webapp/app/components/language/_language.constants.js]: Should this comment be indented to match the code block below?

**Comment by wmarques** [comment]
Thanks I didn't see the indent problem in the alert error directive, I fixed it. I also indented the comment as you suggested, it seems more logic for me if it has the same indent than the code block below.

**Comment by deepu105** [comment]
you should add [ci skip] for indent and formatting only pushes

File: generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    'use strict';
+
+    angular
+        .module('<%=angularAppName%>')
+        .component('jhAlertError', jhAlertError);
+
+    function jhAlertError () {
+        var component = {
+            template: '<div class="alerts" ng-cloak="">' +
+                            '<div ng-repeat="alert in alerts" ng-class="[alert.position, {\'toast\': alert.toast}]">' +
```
Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js]: should this be `vm.alerts`?

Comment (deepu105) [inline generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js]: ill change this

Comment (wmarques) [inline generators/client/templates/src/main/webapp/app/components/alert/_alert-error.directive.js]: Yes it's `vm.alerts`, sorry

I"ll add `[ci skip]` next time for those type of commits, didn't know about that trick !

**Pull request merged**